src/fileReference.ts

```diff
@@ -0,0 +1,42 @@
+import type { Types } from "@graphql-codegen/plugin-helpers";
+import { code, imp, type Code } from "ts-poet";
+import type { Config } from "./types";
+import { generateNearOperationFileImport, type NearOperationFileConfigWithDefaults } from "./nearOperationFileImport";
+
+export type FileReferenceFunction = (typeName: string, documentFile: Types.DocumentFile) => Code;
+
+export function createFileReferenceFunction(config: Config, outputFile?: string): FileReferenceFunction {
```

Comment:
This is the replacement for `maybeImport`. Since there's different behavior between the three main use cases described in the README, I'm returning the proper import function once, then using it throughout.